enhancements/microshift/update-and-rollback.md

```diff
@@ -0,0 +1,255 @@
+
+The backup should consider all data required to save the state of MicroShift such as
+certificates, configuration files, volumes
+
+At every boot, `Greenboot` healthchecks will be executed as described in [this enhancement.](https://github.com/openshift/enhancements/blob/master/enhancements/microshift/microshift-greenboot.md)
+
+If upgrading MicroShift to a new version results in a broken state, `Greenboot` scripts
+will restore the previously saved backup, and perform an atomic rollback to the N-1 state.
+
+It is important to highlight that MicroShift's upgrade and rollback processes need to
+fit well into the mechanisms provided by the OSTree based version of Red Hat Enterprise
+Linux, but it will not own any of these responsabilities. It is the `device administrator`
+duty, to decide the upgrade/rollback workflow end to end.
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [Optional]
+
+#### rpm-ostree Configuration
+
+`rpm-ostree` provides a configuration option to check and stage new updates.
+The file `/etc/rpm-ostreed.conf` provides `AutomaticUpdatePolicy` which defaults to `none`.
+
+This option needs to be set to `stage` which downloads and unpacks the update performing
+any package layering. Only a small amount of work is left to be performed at shutdown time
+via the `ostree-finalize-staged.service` systemd unit.
+
+Furthermore, in order to apply this policy, the `rpm-ostreed-automatic.service` systemd
+service must be enabled.
+
+This configuration is explained in the official documentation for RHEL 9. It can be found
+[here](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/composing_installing_and_managing_rhel_for_edge_images/managing-rhel-for-edge-images_composing-installing-managing-rhel-for-edge-images#enabling-rhel-for-edge-automatic-download-and-staging-of-updates_managing-rhel-for-edge-images).
+
+This step is not strictly part of MicroShift's domain. However, in order to manage edge
+devices at scale, this configuration is recommended.
+
+#### MicroShift's Upgrade Systemd Service
+
+A new systemd service file will be packaged into the MicroShift's main RPM. This systemd
+service will monitor the `/run/ostree/staged-deployment` file that stores the staged update
+deployment info by using the `"ConditionPathExists"` option. This condition will be the
+trigger to run a custom script that saves the MicroShift state into a data backup file.
+
+#### MicroShift Backup
+
+Leveraging the ability to run any custom script before the new OSTree deployment becomes
+active, allow us to perform the following tasks in order:
+* Stop MicroShift systemd service (which does not cause interruption for applications)
+* Calculate an estimate of storage space to avoid getting the device out of disk
+* Save a tarball file of `/var/lib/microshift` to `/var/lib/microshift-backup`
+* Save snapshots of any persistent information store in volumes
```

Comment:
I'm not sure we want MicroShift to be responsible for application data backup.

Reply:
Probably not. MicroShift does not have any volume for its extra running services, right?

Reply:
All of MicroShift's state should be in /var/lib/microshift